ReactNativeClient/lib/file-api-driver-onedrive.js

```diff
@@ -129,19 +130,21 @@ class FileApiDriverOneDrive {
 
 		let response = null;
 
-		try {
-			if (options.source == 'file') {
-				response = await this.api_.exec('PUT', `${this.makePath_(path)}:/content`, null, null, options);
+		if (options.source == 'file') {
+			// We need to check the file size as files > 4 MBs are uploaded in a different way than files < 4 MB (see https://docs.microsoft.com/de-de/onedrive/developer/rest-api/concepts/upload?view=odsp-graph-online)
+			const fileSize = await shim.fsDriver().fileSize(options.path);
+			if (fileSize < 4194304) {
+				path = `${this.makePath_(path)}:/content`;
+				options.fileType = 'small';
 			} else {
-				options.headers = { 'Content-Type': 'text/plain' };
-				response = await this.api_.exec('PUT', `${this.makePath_(path)}:/content`, null, content, options);
-			}
-		} catch (error) {
-			if (error && error.code === 'BadRequest' && error.message === 'Maximum request length exceeded.') {
-				error.code = 'rejectedByTarget';
-				error.message = 'Resource exceeds OneDrive max file size (4MB)';
+				path = `${this.makePath_(path)}:/createUploadSession`;
+				options.fileType = 'big';
+				options.fileSize = fileSize;
```

Comment:
I think these two properties are not necessary as you can derive this info from the path (if it contains "createUploadSession", it needs `uploadBigFile()`. And ideally the file size should be retrieved where it's needed, directly in `uploadBigFile()`.